nengo/builder/operator.py

```diff
@@ -388,7 +388,10 @@ def make_step(self, signals, dt, rng):
 
         if inc:
             def step_copy():
-                dst[dst_slice] += src[src_slice]
+                if dst_slice == Ellipsis:
+                    dst[dst_slice] += src[src_slice]
```

Comment:
dst_slice == Ellipsis if no slice is provided. So we'd think that dst = dst[dst_slice]
Yet,  *dst += src[src_slice]* sometimes fails because "dst referenced before assignment" because of some weird numpy internal